### PR TITLE
Update containerfile and application dependencies to use Fedora 34

### PR DESCRIPTION
**Pull request by mfleader**
We need a non-Docker Hub base image and CentOS 8 is deprecated, so it seemed like a good time to refactor the base image to Fedora 34, update the resulting dependencies, and clean up the documentation and example deployment with Podman.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM docker.io/library/centos:8.2.2004
+FROM quay.io/fedora/fedora:34
```
Comment (chaitanyaenr) [inline Dockerfile:1]: Thoughts on using fedora/fedora:latest? This way we need not change in case we need a new image.

Comment (mfleader) [reply Dockerfile:1]: Pinning a dependency's version with either a version tag or an image SHA will help prevent a breaking change from sneaking into the container image.

**Review by chaitanyaenr — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (chaitanyaenr) [reply Dockerfile:1]: Makes sense.

**Review by mohit-sheth — APPROVED** [review]
LGTM

**Pull request merged**